server.ts

```diff
@@ -0,0 +1,57 @@
+import * as express from 'express';
+import { TodoApp } from './src/todo';
+
+const todo = new TodoApp();
+
+const serv = express();
+serv.use(express.json());
+
+const port = 3000;
+
+serv.get('/', function (req, res) {
+    res.send(todo.listAllToDos());
+});
+
+serv.put('/', function (req, res) {
+    const data = req.body;
+    try {
+        todo.addItem(data.description, data.dueAt);
+        res.sendStatus(200);
+    } catch (e) {
+        if (e.message = 'The date you entered is invalid') {
+            res.status(400).send('The date you entered is invalid');
+        } else {
+            throw e;
+        }
+    }
+});
+
+serv.post(/^\/[0-9]+\/done/, function (req, res) {
+    const id: string = req.path.split('/')[1];
+    try {
+        todo.checkItem(parseInt(id));
+        res.sendStatus(200);
+    } catch (e) {
+        if (e.message = 'No such item exists') {
```

Comment:
Generally speaking it is a bad practice to predicate behavior based on error messages (since these can change easily).  We will talk about error types and how to create your own errors to make this easier.